src/utils.ts

```diff
@@ -59,3 +59,33 @@ export function isRemoteDebuggingEnabled(): boolean {
     !localGlobal.RN$Bridgeless
   );
 }
+
+export function deepEqual(obj1: any, obj2: any) {
+  if (obj1 === obj2) {
+    return true;
+  }
+
+  if (
+    typeof obj1 !== 'object' ||
+    typeof obj2 !== 'object' ||
+    obj1 === null ||
+    obj2 === null
+  ) {
+    return obj1 === obj2;
```

Comment:
This branch always returns `false`, no?

```suggestion
    return false;
```

Reply:
I'm pretty sure you're right.